url.go

```diff
@@ -152,64 +157,111 @@ func ParseReference(url string) (*Reference, error) {
 
 func parseReference(url string) (*Reference, error) {
 	var r Reference
-	i := strings.Index(url, ":")
-	if i >= 0 {
-		r.Schema = url[:i]
-		if r.Schema != "cs" && r.Schema != "local" {
-			return nil, fmt.Errorf("entity URL has invalid schema: %q", url)
+	// Check if we're dealing with a v1 or v2 URL.
+	// TODO: this is a naive check that only works on fully-qualified URLs;
+	// further work will be required for future URLs
+	// XXX Get rid of urlVersion, test based on validStore.MatchString
+	if validStore.MatchString(url) {
```

Comment:
Perhaps split both the arms of this if statement into
two functions (parseNewStyleReference and parseOldStyleReference?).

I'd call parseNewStyleReference iff the url starts with https?: - then
the error messages will be sensible if the url is a new style URL
but not entirely well formed.